qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -739,7 +739,7 @@ class Gaussian(metaclass=_PulseType):
     def __new__(
         cls,
         duration: Union[int, ParameterExpression],
-        amp: Union[complex, float, ParameterExpression],
+        amp: Union[float, ParameterExpression],
```

Comment:
I'm not sure when we should update the type hint, because complex is still acceptable with warning. @Eric-Arellano ?

We have convenient alias for this type.
https://github.com/Qiskit/qiskit-terra/blob/e9f8b7c50968501e019d0cb426676ac606eb5a10/qiskit/circuit/parameterexpression.py#L30

Reply:
I wasn't sure either. I thought it will serve as an additional hint for the user.

I'll swap in the `ParameterValueType`. Thanks.

Reply:
It depends on what user experience you want for your end users who use MyPy. Do you want them to know about the deprecation because of both MyPy and runtime warnings? I think it is a good idea! They could ignore the MyPy warning with a `# type: ignore` comment. But some people could argue it's breaking our deprecation policy? 

cc @1ucian0 for guidance.

Reply:
I dont think the deprecation policy is explicit about this situation. Removing the type might also warn IDE users about the change. It looks to me like the right way.

Reply:
It is an interesting question. I think changing the type now makes sense as well, if we think about the deprecation policy as protecting running code but not CI tools. For projects in the Qiskit ecosystem, the deprecation process already breaks CI because we generally set up the tests so they fail if a deprecation warning is emitted, so also failing mypy if a deprecated type is used makes sense. Also, beyond mypy, the type hint is used for documenting the expected type and we would not want to encourage users to keep using the deprecated type.

Reply:
Thanks for the input!